File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        ans = 0
        n = len(nums)
        hmap={}
        for i in range(len(nums)):
           hmap[nums[i]] =hmap.get(nums[i], 0) + 1
        for key, value in hmap.items():
            if value > n//2:
                return key
        return -1
        

            
        
        

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int,int>has;
        int n = nums.size();
        for(int i = 0; i<n;i++){
            has[nums[i]]++;
        }
        for(auto it:has){
        if (it.second > (n/2)){
        return it.first;
        }
        }return -1;
    }
};